mod/state-transition/pkg/core/state/statedb.go

```diff
@@ -228,6 +219,12 @@ func (s *StateDB[
 		return withdrawals, nil
 	}
 
+	if s.cs.DepositEth1ChainID() == spec.BoonetEth1ChainID &&
+		slot.Unwrap() >= spec.BoonetFork2Height {
```

Comment:
I think this is wrong.
We should add the inflation withdrawal by default and do only two exceptions:
- Bartio
- Boonet pre Fork 2

Instead here any network that is not Cartio post fork 2 won't make the inflation withdrawal, which I think it's wrong